controllers/cluster_controller_phases.go

```diff
@@ -175,6 +175,12 @@ func (r *ClusterReconciler) reconcileInfrastructure(ctx context.Context, cluster
 		}
 	}
 
+	// Get and parse Status.FailureDomains from the infrastructure provider.
+	if err := util.UnstructuredUnmarshalField(infraConfig, &cluster.Status.FailureDomains, "spec", "failureDomains"); err != nil {
+		return errors.Wrapf(err, "failed to retrieve Status.FailureDomains from infrastructure provider for Cluster %q in namespace %q",
+			cluster.Name, cluster.Namespace)
```

Comment:
Should this swallow `ErrUnstructuredFieldNotFound` to allow for optional support for Infrastructure providers?

Reply:
Yep, sounds good

Reply:
This looks to still be returning ErrUnstructuredFieldNotFound back up, are we making the FailureDomain field mandatory for infrastructure providers now?

Reply:
I added `err != util.ErrUnstructuredFieldNotFound` to the if condition, unless I'm mistaken, that should be sufficient?

Reply:
Ah, indeed missed that diff due to the scrolling, thanks :)